### PR TITLE
Add build container make target

### PR DESCRIPTION
**Pull request by spartan0x117**
Adds a new make target and modifies `buf-generate`:
- `build-container` uses `golang:1.22.1` (this should be updated as we update the version in `go.mod`) to install `protoc-gen-go` and `bufbuild/buf` container to create a new image that can be used to `buf generate`
- `buf-generate` now uses the previously built image to `buf generate` and then copy the resulting generated code to the host.

This should make it easier to regenerate based on `.proto` definitions, without having to worry about differing versions of the `buf` cli tool or `protoc-gen-go`.

### CODE REVIEW

**Comment by captncraig** [comment]
Some makefiles (like alloy) have an env var to build in container, rather than a separate target. I don't have a strong opinion, but want to make sure we look at it for consistency.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Stage 1: Use a Golang image to get protoc-gen-go
+FROM golang:latest AS go_builder
```
Comment (erikbaranowski) [inline Dockerfile:2]: let's use the same specific golang version here so we don't drift at all

**Review by erikbaranowski — APPROVED** [review]
This is nice to see with a docker container. I'd love to see some CI eventually that makes sure the generated code hasn't drifted in a PR using the docker container.

File: Dockerfile (hunk 1 above)
Comment (spartan0x117) [reply Dockerfile]: Good point, done!

**Pull request merged**